File: Component_DiceGame/DiceGame_csharp/ComponentDiceGame/view/EngConsole.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Console.view
{
	class EngConsole : Console
	{

		override public void DisplayInstructions()
		{
			System.Console.Clear();
			String message = "Welcome to the cool Dice Game. Press any Key to play, or q to Quit";
			System.Console.WriteLine(message);
		}

		override public bool WantsToPlay()
		{
			return System.Console.ReadKey().KeyChar != 'q';
		}

		override public void DisplayResult(bool a_isWinner)
		{
			if (a_isWinner)
			{
				System.Console.WriteLine("You are a winner!");
			}
			else
			{
				System.Console.WriteLine("Sorry you lost!");
			}
		}
	}
}

File: MVC_O_DiceGame/DiceGame_ruby/view/engConsole.rb
module View
  class EngConsole < Console

    QUIT_ON_INPUT  = "q"

    def display_instructions
      system "clear"
      puts "Welcome to the cool Dice Game. Press any Key to play, or #{QUIT_ON_INPUT} to Quit"
    end

    def wants_to_quit?
      gets == "#{QUIT_ON_INPUT}\n"
    end

    def display_result player_won
        if player_won
          puts "You are a winner!"
        else
          puts "Sorry you lost!"
        end
    end

  end
end
